packaging/hip-targets-release.cmake

```diff
@@ -12,7 +12,7 @@ get_filename_component(_IMPORT_PREFIX "${_DIR}/../../../" REALPATH)
 
 # Import target "hip::hip_hcc_static" for configuration "Release"
 set_property(TARGET hip::hip_hcc_static APPEND PROPERTY IMPORTED_CONFIGURATIONS RELEASE)
-if(HIP_COMPILER STREQUAL "hcc" OR HIP_COMPILER STREQUAL "clang")
+if(CMAKE_CXX_COMPILER MATCHES "(.*)hcc" OR CMAKE_CXX_COMPILER MATCHES "(.*)clang(.*)")
 set_target_properties(hip::hip_hcc_static PROPERTIES
   IMPORTED_LINK_INTERFACE_LANGUAGES_RELEASE "CXX"
   IMPORTED_LINK_INTERFACE_LIBRARIES_RELEASE "hc_am"
```

Comment:
The following `hip::hip_hcc_static` properties are unconditional and might be brought out of `if() ... else()`; `else()` is exhaustive:
```cmake
IMPORTED_LINK_INTERFACE_LANGUAGES_RELEASE "CXX"
IMPORTED_LOCATION_RELEASE "${_IMPORT_PREFIX}/lib/libhip_hcc_static.a"
```